paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/source/statistics/FileStoreTableStatisticsTestBase.java

```diff
@@ -0,0 +1,181 @@
+        Assertions.assertThat(scanSource.reportStatistics().getRowCount()).isEqualTo(9L);
+        // TODO validate column statistics
+    }
+
+    @Test
+    public void testTableStreamingStatistics() throws Exception {
+        FileStoreTable table = writeData();
+        DataTableSource streamSource = new DataTableSource(identifier, table, true, null, null);
+        Assertions.assertThat(streamSource.reportStatistics()).isEqualTo(TableStats.UNKNOWN);
+    }
+
+    @Test
+    public void testTableFilterPartitionStatistics() throws Exception {
+        FileStoreTable table = writeData();
+        PredicateBuilder builder = new PredicateBuilder(table.schema().logicalRowType());
+        DataTableSource partitionFilterSource =
+                new DataTableSource(
+                        identifier,
+                        table,
+                        false,
+                        null,
+                        null,
+                        builder.equal(0, 1),
+                        null,
+                        null,
+                        null);
+        Assertions.assertThat(partitionFilterSource.reportStatistics().getRowCount()).isEqualTo(5L);
+        // TODO validate column statistics
+    }
+
+    @Test
+    public void testTableFilterKeyStatistics() throws Exception {
+        FileStoreTable table = writeData();
+        PredicateBuilder builder = new PredicateBuilder(table.schema().logicalRowType());
+        DataTableSource keyFilterSource =
+                new DataTableSource(
+                        identifier,
+                        table,
+                        false,
+                        null,
+                        null,
+                        builder.equal(1, 50),
+                        null,
+                        null,
+                        null);
+        Assertions.assertThat(keyFilterSource.reportStatistics().getRowCount()).isEqualTo(2L);
+        // TODO validate column statistics
+    }
+
+    @Test
+    public void testTableFilterValueStatistics() throws Exception {
+        FileStoreTable table = writeData();
+        PredicateBuilder builder = new PredicateBuilder(table.schema().logicalRowType());
+        DataTableSource keyFilterSource =
+                new DataTableSource(
+                        identifier,
+                        table,
+                        false,
+                        null,
+                        null,
+                        builder.greaterThan(2, 500L),
+                        null,
+                        null,
+                        null);
+        Assertions.assertThat(keyFilterSource.reportStatistics().getRowCount()).isEqualTo(4L);
+        // TODO validate column statistics
+    }
+
+    protected FileStoreTable writeData() throws Exception {
+        FileStoreTable table = createStoreTable();
+
+        TableWriteImpl write = table.newWrite(commitUser);
```

Comment:
Use `TableWriteImpl<?>` to avoid `Raw use` warning in IDE.

Reply:
Thanks @yuzelin DONE